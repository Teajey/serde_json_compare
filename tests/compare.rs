use serde_json::json;

#[test]
fn objects() {
    let a = json!({
      "1": "2",
      "3": 4,
      "4": [4, "5"],
      "5": [4, "5"],
      "-20": [4, "5"],
      "numbero": 1
    })
    .as_object()
    .unwrap()
    .clone();
    let b = json!({
      "1": "2",
      "3": "4",
      "4": [4, "6", 7],
      "5": [4, "7"],
      "100": [4, "7"],
      "numbero": 7
    })
    .as_object()
    .unwrap()
    .clone();

    let diff = serde_json_compare::objects(a, b);

    let diff = serde_json::to_value(diff).expect("couldn't serialize diff");

    insta::assert_snapshot!(serde_json::to_string_pretty(&diff).expect("couldn't pretty"));
}
