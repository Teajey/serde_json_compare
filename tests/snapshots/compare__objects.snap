---
source: tests/compare.rs
expression: "serde_json::to_string_pretty(&diff).expect(\"couldn't pretty\")"
---
{
  "-20": "MissingKey",
  "100": "ExtraKey",
  "3": {
    "TypeMismatch": [
      "Number",
      "String"
    ]
  },
  "4": {
    "Array": {
      "LengthMismatch": [
        2,
        3
      ]
    }
  },
  "5": {
    "Array": {
      "MismatchingElements": {
        "1": {
          "ValueMismatch": [
            "5",
            "7"
          ]
        }
      }
    }
  },
  "numbero": {
    "ValueMismatch": [
      1,
      7
    ]
  }
}
